examples/8a_benchmark_write_parallel.cpp

```diff
@@ -1131,6 +1144,19 @@ unsigned int AbstractPattern::getNumBlocks()
     return m_InRankMeshLayout.size();
 }
 
+inline openPMD::Iteration
+AbstractPattern::GetIteration(Series &series, int const iteration) const
+{
+    if (m_Input.m_Encoding == openPMD::IterationEncoding::variableBased)
+    {
+        return series.writeIterations()[iteration];
+    }
+    else
+    {
+        return series.iterations[iteration];
```

Comment:
Is there any reason to not use `writeIterations()` here? It should also work in group-based and file-based encoding

Reply:
The file/group based  were not using begin/end steps that is why they were not using writeIterations.

As for json, it is convenient to put all the parameters in one input file. That's why I added these options.

Reply:
> The file/group based were not using begin/end steps that is why they were not using writeIterations.

`readIterations()` should work even if you did not use `writeIterations()`, see [this test](https://github.com/openPMD/openPMD-api/blob/5a65f6df28a1f510d0ae242293d2d410aa32cda5/test/SerialIOTest.cpp#L5029). If it does not, then it's a bug.

> 
> As for json, it is convenient to put all the parameters in one input file. That's why I added these options.

That's reasonable yup

Reply:
ok, then let's add it here, because then read works also with data generated with steps/streaming

Reply:
Can we remove the `series.iterations` branch then? Since this file is one of the openPMD-api examples, I'm reluctant to advertize that users need to switch between both APIs.

Reply:
> ok, then let's add it here, because then read works also with data generated with steps/streaming?

sorry, add what where?  yes read works with variable based files.

Reply:
> Can we remove the `series.iterations` branch then? Since this file is one of the openPMD-api examples, I'm reluctant to advertize that users need to switch between both APIs.

OK, let me see to it.

Reply:
> `readIterations()` should work even if you did not use `writeIterations()`, see [this test](https://github.com/openPMD/openPMD-api/blob/5a65f6df28a1f510d0ae242293d2d410aa32cda5/test/SerialIOTest.cpp#L5029). If it does not, then it's a bug.

If I just use writeIterations for all encodings, it does not work.

Reply:
I added code and make sure  BP5 must be associated with variable based iteration. Now can safely use  writeIteration() only.

Reply:
Ok, as a last step - do you still need `AbstractPattern::GetIteration` for the one-liner?

Reply:
I think can leave it there. Might be flexible for later usage.

Reply:
we really dont need it anymore and this is an example, where we want to encourage best current usage to users.

we cann sdd helpers when we really need them

Reply:
That's fine.

Reply:
So the bug was only BP5 related? If yes, then it should be fixed with #1215, and I'd prefer that fix in favor of a workaround here

Reply:
> So the bug was only BP5 related? If yes, then it should be fixed with #1215, and I'd prefer that fix in favor of a workaround here

yes, BP5 enforces begin/end steps()

Reply:
I think so